owncloudApp/src/main/java/com/owncloud/android/presentation/files/createshortcut/CreateShortcutDialogFragment.kt

```diff
@@ -0,0 +1,154 @@
+/**
+ * ownCloud Android client application
+ *
+ * @author Aitor Ballesteros Pavón
+ *
+ * Copyright (C) 2024 ownCloud GmbH.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package com.owncloud.android.presentation.files.createshortcut
+
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.core.widget.doOnTextChanged
+import androidx.fragment.app.DialogFragment
+import com.owncloud.android.R
+import com.owncloud.android.databinding.CreateShortcutDialogBinding
+import com.owncloud.android.domain.files.model.OCFile
+
+class CreateShortcutDialogFragment : DialogFragment() {
+    private lateinit var parentFolder: OCFile
+    private lateinit var createShortcutListener: CreateShortcutListener
+    private var _binding: CreateShortcutDialogBinding? = null
+    private var isCreateShortcutButtonEnable = false
+    private var isValidUrl = false
+    private var isValidFileName = false
+    private var hasForbiddenCharacters = false
+    private var hasMaxCharacters = false
+    private var hasEmptyValue = false
```

Comment:
Do we really need all of these variables to be class properties? Couldn't they be local variables in the scope of the methods where they are needed?